src/software/new_geom/angle.h

```diff
@@ -0,0 +1,595 @@
+{
+    return std::sin(toRadians());
+}
+
+inline double Angle::cos() const
+{
+    return std::cos(toRadians());
+}
+
+inline double Angle::tan() const
+{
+    return std::tan(toRadians());
+}
+
+inline constexpr Angle Angle::clamp() const
+{
+    return remainder(Angle::full());
+}
+
+inline constexpr Angle Angle::minDiff(const Angle &other) const
+{
+    return (*this - other).clamp().abs();
+}
+
+inline constexpr Angle Angle::angleMod() const
+{
+    return remainder(Angle::full());
+}
+
+inline constexpr Angle::Angle(double rads) : rads(rads) {}
+
+inline constexpr Angle operator-(const Angle &angle)
+{
+    return Angle::ofRadians(-angle.toRadians());
+}
+
+inline constexpr Angle operator+(const Angle &x, const Angle &y)
+{
+    return Angle::ofRadians(x.toRadians() + y.toRadians());
+}
+
+inline constexpr Angle operator-(const Angle &x, const Angle &y)
+{
+    return Angle::ofRadians(x.toRadians() - y.toRadians());
+}
+
+inline constexpr Angle operator*(const Angle &angle, double scale)
+{
+    return Angle::ofRadians(angle.toRadians() * scale);
+}
+
+inline constexpr Angle operator*(double scale, const Angle &angle)
+{
+    return Angle::ofRadians(scale * angle.toRadians());
+}
+
+inline constexpr Angle operator/(const Angle &angle, double divisor)
+{
+    return Angle::ofRadians(angle.toRadians() / divisor);
+}
+
+inline constexpr double operator/(const Angle &x, const Angle &y)
+{
+    return x.toRadians() / y.toRadians();
+}
+
+inline Angle &operator+=(Angle &x, const Angle &y)
+{
+    return x = x + y;
+}
+
+inline Angle &operator-=(Angle &x, const Angle &y)
+{
+    return x = x - y;
+}
+
+inline Angle &operator*=(Angle &angle, double scale)
+{
+    return angle = angle * scale;
+}
+
+inline Angle &operator/=(Angle &angle, double divisor)
+{
+    return angle = angle / divisor;
+}
+
+inline constexpr bool operator<(const Angle &x, const Angle &y)
+{
+    return x.toRadians() < y.toRadians();
+}
+
+inline constexpr bool operator>(const Angle &x, const Angle &y)
+{
+    return x.toRadians() > y.toRadians();
+}
+
+inline constexpr bool operator<=(const Angle &x, const Angle &y)
+{
+    return x.toRadians() <= y.toRadians();
+}
+
+inline constexpr bool operator>=(const Angle &x, const Angle &y)
+{
+    return x.toRadians() >= y.toRadians();
+}
+
+inline bool operator==(const Angle &x, const Angle &y)
+{
+    Angle diff = x.angleMod().minDiff(y.angleMod());
+    return diff.toRadians() <= GeomConstants::EPSILON;
+}
+
+inline constexpr bool operator!=(const Angle &x, const Angle &y)
+{
+    return x.toRadians() != y.toRadians();
+}
+
+inline std::ostream &operator<<(std::ostream &os, const Angle &a)
+{
+    os << a.toRadians() << "R";
+    return os;
+}
+
+// We also use variables of type 'Angle' to represent angular velocities, since they
+// are essentially represented the same. This typedef allows us to refer to Angles as
+// AngularVelocities, which makes the interfaces more intuitive.
+typedef Angle AngularVelocity;
```

Comment:
can we put this in a `angular_velocity.h` file since it's not really part of the `Angle` class?